CameraOverlay/overlay_all_stats.py

```diff
@@ -132,18 +134,22 @@ def on_message(self, client, userdata, msg):
 				self.data["reed_distance"] = 0
 				self.data["count"] = 0
 		
-		elif topic == str(topics.DAShboard.recieve_message):
+		elif ((topic == str(topics.DAShboard.receive_message)) & self.message_received_flag == False):
+			self.message_received_time = time.time()
 			message = msg.payload.decode("utf-8")
 
-			# reset counter once message is recieved 
-			self.frame_counter = 0  
-
 			# Display Message
-			self.message_canvas.draw_text("{0}".format(message), (190, self.text_height * 5), size= 1, colour=Colour.red)
-		
-		if self.frame_counter >= 70:
+			self.message_canvas.draw_text(message, (190, self.text_height * 5), size= 1, colour=Colour.red)
+
+			# Set message_received_flag to True once a message is received
+			self.message_received_flag  = True
+
+		if (time.time() - self.message_received_time) >= 5:
 			self.message_canvas.clear()
 
+			# Set message_received_flag to True once displayed message is cleared
```

Comment:
Sets to `False`, not `True`. The comment (and the related one above) is probably superfluous with the variable name change anyway.